specter-registry/src/testmod/java/dev/spiritstudios/testmod/SpecterRegistryTestCommand.java

```diff
@@ -0,0 +1,96 @@
+package dev.spiritstudios.testmod;
+
+import com.google.common.collect.Streams;
+import com.mojang.brigadier.Command;
+import com.mojang.brigadier.CommandDispatcher;
+import com.mojang.brigadier.StringReader;
+import com.mojang.brigadier.arguments.ArgumentType;
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.suggestion.Suggestions;
+import com.mojang.brigadier.suggestion.SuggestionsBuilder;
+import com.mojang.datafixers.util.Pair;
+import com.mojang.serialization.JsonOps;
+import dev.spiritstudios.specter.api.registry.attachment.Attachment;
+import dev.spiritstudios.specter.impl.registry.attachment.AttachmentHolder;
+import dev.spiritstudios.specter.impl.registry.attachment.data.AttachmentResource;
+import net.minecraft.command.CommandRegistryAccess;
+import net.minecraft.command.CommandSource;
+import net.minecraft.registry.Registries;
+import net.minecraft.registry.Registry;
+import net.minecraft.registry.RegistryKey;
+import net.minecraft.server.command.CommandManager;
+import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.text.Text;
+import net.minecraft.util.Identifier;
+
+import java.util.concurrent.CompletableFuture;
+
+public class SpecterRegistryTestCommand {
+	public static class RegistryArgumentType implements ArgumentType<Registry<?>> {
+		@Override
+		public Registry<?> parse(StringReader reader) throws CommandSyntaxException {
+			return Registries.ROOT.get(Identifier.fromCommandInput(reader));
+		}
+
+		@Override
+		public <S> CompletableFuture<Suggestions> listSuggestions(CommandContext<S> context, SuggestionsBuilder builder) {
+			return context.getSource() instanceof CommandSource
+				? CommandSource.suggestIdentifiers(Registries.ROOT.stream().map(r -> r.getKey().getValue()), builder)
+				: Suggestions.empty();
+		}
+	}
+
+	public static class AttachmentArgumentType implements ArgumentType<Identifier> {
+
+		protected final String registryArg;
+		public AttachmentArgumentType(String registryArg) {
+			this.registryArg = registryArg;
+		}
+
+		@Override
+		public Identifier parse(StringReader reader) throws CommandSyntaxException {
+			return Identifier.fromCommandInput(reader);
+		}
+
+		@Override
+		public <S> CompletableFuture<Suggestions> listSuggestions(CommandContext<S> context, SuggestionsBuilder builder) {
+			Registry<?> registry = context.getArgument(registryArg, Registry.class);
+			CommandSource.suggestIdentifiers(registry.getKeys().stream().map(RegistryKey::getValue), builder);
+			return CommandSource.suggestIdentifiers(registry.getKeys().stream().map(RegistryKey::getValue), builder);
+		}
+
+	}
+
+	private static Attachment<?, ?> getAttachmentFromContext(CommandContext<ServerCommandSource> ctx) {
```

Comment:
Once again minor style thing but i generally use `context` instead of `ctx`. You seem to have done that above so it would be nice for consistency.